Tella/Data/Networking/Upload/UploadService.swift

```diff
@@ -64,6 +63,23 @@ class UploadService: NSObject {
         activeOperations.append(operation)
         uploadQueue.addOperation(operation)
         operation.startUploadReportAndFiles()
+
+        // Display information toast
+        operation.response.sink { completion in
+        } receiveValue: { response in
+            switch response {
+            case .finish(let isAutoDelete , let title):
```

Comment:
could we extract the logic into separate functions for better readability? Something like this:

```
func handleResponse(_ response: Response) {
    switch response {
    case .finish(let isAutoDelete, let title):
        displaySubmittedToast(title: title)
        if isAutoDelete {
            displayDeletedToast(title: title)
        }
    default:
        break
    }
}

func displaySubmittedToast(title: String?) {
    let message = String(format: LocalizableReport.reportSubmittedToast.localized, title ?? "")
    Toast.displayToast(message: message)
}

func displayDeletedToast(title: String?) {
    DispatchQueue.main.asyncAfter(deadline: .now() + 3) {
        let message = String(format: LocalizableReport.reportDeletedToast.localized, title ?? "")
        Toast.displayToast(message: message)
    }
}

operation.response
    .sink(receiveCompletion: { _ in })
    { response in
        handleResponse(response)
    }
    .store(in: &subscribers)
    ```

Reply:
I did your suggestion and I centralised the code into a function.